schema.json

```diff
@@ -0,0 +1,264 @@
+{
+    "workshop": {
+	"type": "dict",
+	"required": true,
+	"schema": {
+	    "schema": {
+		"type": "dict",
+		"required": true,
+		"schema": {
+		    "version": {
+			"allowed": [
+			    "2020.03.02"
```

Comment:
My only concern with this type of versioning is that it's really hard to tell how many versions have occurred between 1 version and another.  Using whole numbers avoids this.  However, it is not a big enough concern to use something else.

Reply:
Just a general discussion, does it matter how many have occurred between versions?

One subtle thing here that isn't documented (yet) is that I made the allowed versions a list.  If changes to the schema are made that result in the addition of optional properties then the new version can simply be added to the list.  When an incompatible change is introduced then the list needs to be scrubbed with only the new version allowed.